ouroboros-consensus/test-util/Test/Util/QuickCheck.hs

```diff
@@ -21,3 +26,15 @@ x `ge` y = counterexample (show x ++ " < " ++ show y) $ x >= y
 -- | Like '<', but prints a counterexample when it fails.
 lt :: (Ord a, Show a) => a -> a -> Property
 x `lt` y = counterexample (show x ++ " >= " ++ show y) $ x < y
+
+{-------------------------------------------------------------------------------
+  Improved variants
+-------------------------------------------------------------------------------}
+
+-- | Generates one of the given values. The input list must be non-empty.
+--
+-- NOTE unlike the standard @elements@, this variant has a 'HasCallStack'
+-- constraint, which makes debugging the 'error' much easier.
+elements :: HasCallStack => [a] -> Gen a
+elements [] = error "QuickCheck.elements used with empty list"
```

Comment:
Perhaps an irrelevant nitpick but, because this could be confused with the actual module from the `quickcheck` package, I'd be inclined to be more explicit here:

```suggestion
elements [] = error "Test.Util.QuickCheck.elements used with empty list"
```